## Pre-Processing

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import cross_val_score, train_test_split
from sklearn.metrics import r2_score, mean_squared_error
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler, PolynomialFeatures 
from sklearn import metrics


import statsmodels.api as sm
from statsmodels.stats.outliers_influence import variance_inflation_factor

In [3]:
train = pd.read_csv('../datasets/train.csv')

kaggle = pd.read_csv('../datasets/test 2.csv')

train.head()

id        pid  ms_subclass ms_zoning  lot_frontage  lot_area street  \
0  109  533352170           60        RL           0.0     13517   Pave   
1  544  531379050           60        RL          43.0     11492   Pave   
2  153  535304180           20        RL          68.0      7922   Pave   
3  318  916386060           60        RL          73.0      9802   Pave   
4  255  906425045           50        RL          82.0     14235   Pave   

  lot_shape land_contour utilities  ... square_footage bldg_type_1Fam  \
0       IR1          Lvl    AllPub  ...         2204.0              1   
1       IR1          Lvl    AllPub  ...         3035.0              1   
2       Reg          Lvl    AllPub  ...         2114.0              1   
3       Reg          Lvl    AllPub  ...         1828.0              1   
4       IR1          Lvl    AllPub  ...         2121.0              1   

  bldg_type_2fmCon bldg_type_Duplex bldg_type_Twnhs bldg_type_TwnhsE  \
0                0                0               0                0   
1                0                0               0                0   
2                0                0               0                0   
3                0                0               0                0   
4                0                0               0                0   

   kitchen_qual_Ex  kitchen_qual_Fa  kitchen_qual_Gd  kitchen_qual_TA  
0                0                0                1                0  
1                0                0                1                0  
2                0                0                1                0  
3                0                0                0                1  
4                0                0                0                1  

[5 rows x 84 columns]

In [5]:
#One - hot encoded
#train = pd.get_dummies(train, columns = ['bldg_type'])
#train.set_index(train.index, inplace = True)

In [6]:
train

id        pid  ms_subclass ms_zoning  lot_frontage  lot_area street  \
0      109  533352170           60        RL           0.0     13517   Pave   
1      544  531379050           60        RL          43.0     11492   Pave   
2      153  535304180           20        RL          68.0      7922   Pave   
3      318  916386060           60        RL          73.0      9802   Pave   
4      255  906425045           50        RL          82.0     14235   Pave   
...    ...        ...          ...       ...           ...       ...    ...   
2044  1587  921126030           20        RL          79.0     11449   Pave   
2045   785  905377130           30        RL           0.0     12342   Pave   
2046   916  909253010           50        RL          57.0      7558   Pave   
2047   639  535179160           20        RL          80.0     10400   Pave   
2048    10  527162130           60        RL          60.0      7500   Pave   

     lot_shape land_contour utilities  ... square_footage bldg_type_1Fam  \
0          IR1          Lvl    AllPub  ...         2204.0              1   
1          IR1          Lvl    AllPub  ...         3035.0              1   
2          Reg          Lvl    AllPub  ...         2114.0              1   
3          Reg          Lvl    AllPub  ...         1828.0              1   
4          IR1          Lvl    AllPub  ...         2121.0              1   
...        ...          ...       ...  ...            ...            ...   
2044       IR1          HLS    AllPub  ...         3612.0              1   
2045       IR1          Lvl    AllPub  ...         1722.0              1   
2046       Reg          Bnk    AllPub  ...         2809.0              1   
2047       Reg          Lvl    AllPub  ...         2400.0              1   
2048       Reg          Lvl    AllPub  ...         2798.0              1   

     bldg_type_2fmCon bldg_type_Duplex bldg_type_Twnhs bldg_type_TwnhsE  \
0                   0                0               0                0   
1                   0                0               0                0   
2                   0                0               0                0   
3                   0                0               0                0   
4                   0                0               0                0   
...               ...              ...             ...              ...   
2044                0                0               0                0   
2045                0                0               0                0   
2046                0                0               0                0   
2047                0                0               0                0   
2048                0                0               0                0   

      kitchen_qual_Ex  kitchen_qual_Fa  kitchen_qual_Gd  kitchen_qual_TA  
0                   0                0                1                0  
1                   0                0                1                0  
2                   0                0                1                0  
3                   0                0                0                1  
4                   0                0                0                1  
...               ...              ...              ...              ...  
2044                0                0                1                0  
2045                0                0                0                1  
2046                0                0                0                1  
2047                0                0                0                1  
2048                0                0                1                0  

[2049 rows x 84 columns]

In [7]:
#train = pd.get_dummies(train, columns = ['kitchen_qual'])
#train.set_index(train.index, inplace = True)

In [8]:
train

id        pid  ms_subclass ms_zoning  lot_frontage  lot_area street  \
0      109  533352170           60        RL           0.0     13517   Pave   
1      544  531379050           60        RL          43.0     11492   Pave   
2      153  535304180           20        RL          68.0      7922   Pave   
3      318  916386060           60        RL          73.0      9802   Pave   
4      255  906425045           50        RL          82.0     14235   Pave   
...    ...        ...          ...       ...           ...       ...    ...   
2044  1587  921126030           20        RL          79.0     11449   Pave   
2045   785  905377130           30        RL           0.0     12342   Pave   
2046   916  909253010           50        RL          57.0      7558   Pave   
2047   639  535179160           20        RL          80.0     10400   Pave   
2048    10  527162130           60        RL          60.0      7500   Pave   

     lot_shape land_contour utilities  ... square_footage bldg_type_1Fam  \
0          IR1          Lvl    AllPub  ...         2204.0              1   
1          IR1          Lvl    AllPub  ...         3035.0              1   
2          Reg          Lvl    AllPub  ...         2114.0              1   
3          Reg          Lvl    AllPub  ...         1828.0              1   
4          IR1          Lvl    AllPub  ...         2121.0              1   
...        ...          ...       ...  ...            ...            ...   
2044       IR1          HLS    AllPub  ...         3612.0              1   
2045       IR1          Lvl    AllPub  ...         1722.0              1   
2046       Reg          Bnk    AllPub  ...         2809.0              1   
2047       Reg          Lvl    AllPub  ...         2400.0              1   
2048       Reg          Lvl    AllPub  ...         2798.0              1   

     bldg_type_2fmCon bldg_type_Duplex bldg_type_Twnhs bldg_type_TwnhsE  \
0                   0                0               0                0   
1                   0                0               0                0   
2                   0                0               0                0   
3                   0                0               0                0   
4                   0                0               0                0   
...               ...              ...             ...              ...   
2044                0                0               0                0   
2045                0                0               0                0   
2046                0                0               0                0   
2047                0                0               0                0   
2048                0                0               0                0   

      kitchen_qual_Ex  kitchen_qual_Fa  kitchen_qual_Gd  kitchen_qual_TA  
0                   0                0                1                0  
1                   0                0                1                0  
2                   0                0                1                0  
3                   0                0                0                1  
4                   0                0                0                1  
...               ...              ...              ...              ...  
2044                0                0                1                0  
2045                0                0                0                1  
2046                0                0                0                1  
2047                0                0                0                1  
2048                0                0                1                0  

[2049 rows x 84 columns]

***Dummy Added***

Split the type of building into dummy variables to show which type of buildings will prove most impacted. This was also done with kitchen quality to see how quality type will effect salesprice.

In [9]:
print(train)

        id        pid  ms_subclass ms_zoning  lot_frontage  lot_area street  \
0      109  533352170           60        RL           0.0     13517   Pave   
1      544  531379050           60        RL          43.0     11492   Pave   
2      153  535304180           20        RL          68.0      7922   Pave   
3      318  916386060           60        RL          73.0      9802   Pave   
4      255  906425045           50        RL          82.0     14235   Pave   
...    ...        ...          ...       ...           ...       ...    ...   
2044  1587  921126030           20        RL          79.0     11449   Pave   
2045   785  905377130           30        RL           0.0     12342   Pave   
2046   916  909253010           50        RL          57.0      7558   Pave   
2047   639  535179160           20        RL          80.0     10400   Pave   
2048    10  527162130           60        RL          60.0      7500   Pave   

     lot_shape land_contour utilities  ... square_f

In [10]:
train.head()

id        pid  ms_subclass ms_zoning  lot_frontage  lot_area street  \
0  109  533352170           60        RL           0.0     13517   Pave   
1  544  531379050           60        RL          43.0     11492   Pave   
2  153  535304180           20        RL          68.0      7922   Pave   
3  318  916386060           60        RL          73.0      9802   Pave   
4  255  906425045           50        RL          82.0     14235   Pave   

  lot_shape land_contour utilities  ... square_footage bldg_type_1Fam  \
0       IR1          Lvl    AllPub  ...         2204.0              1   
1       IR1          Lvl    AllPub  ...         3035.0              1   
2       Reg          Lvl    AllPub  ...         2114.0              1   
3       Reg          Lvl    AllPub  ...         1828.0              1   
4       IR1          Lvl    AllPub  ...         2121.0              1   

  bldg_type_2fmCon bldg_type_Duplex bldg_type_Twnhs bldg_type_TwnhsE  \
0                0                0               0                0   
1                0                0               0                0   
2                0                0               0                0   
3                0                0               0                0   
4                0                0               0                0   

   kitchen_qual_Ex  kitchen_qual_Fa  kitchen_qual_Gd  kitchen_qual_TA  
0                0                0                1                0  
1                0                0                1                0  
2                0                0                1                0  
3                0                0                0                1  
4                0                0                0                1  

[5 rows x 84 columns]

In [11]:
train.columns

Index(['id', 'pid', 'ms_subclass', 'ms_zoning', 'lot_frontage', 'lot_area',
       'street', 'lot_shape', 'land_contour', 'utilities', 'lot_config',
       'land_slope', 'neighborhood', 'condition_1', 'condition_2',
       'house_style', 'overall_qual', 'overall_cond', 'year_built',
       'year_remod/add', 'roof_style', 'roof_matl', 'exterior_1st',
       'exterior_2nd', 'mas_vnr_type', 'mas_vnr_area', 'exter_qual',
       'exter_cond', 'foundation', 'bsmt_qual', 'bsmt_cond', 'bsmt_exposure',
       'bsmtfin_type_1', 'bsmtfin_sf_1', 'bsmtfin_type_2', 'bsmtfin_sf_2',
       'bsmt_unf_sf', 'total_bsmt_sf', 'heating', 'heating_qc', 'central_air',
       'electrical', '1st_flr_sf', '2nd_flr_sf', 'low_qual_fin_sf',
       'gr_liv_area', 'bsmt_full_bath', 'bsmt_half_bath', 'full_bath',
       'half_bath', 'bedroom_abvgr', 'kitchen_abvgr', 'totrms_abvgrd',
       'functional', 'fireplaces', 'garage_type', 'garage_yr_blt',
       'garage_finish', 'garage_cars', 'garage_area', 'garage_qual',
 

## Model 1

Basic qualities everyone looks at.

In [12]:
#train.to_csv('./datasets/train.csv', index = False)

In [40]:
X = train[['year_built', 'overall_cond', 'full_bath', 'garage_area', 'square_footage', 'lot_frontage', 'lot_area', 'yr_sold', 'overall_qual']]

y = train['saleprice']

In [41]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 44)

In [42]:
lr = LinearRegression()

In [43]:
cross_val_score(lr, X_train, y_train).mean()

0.7678629136059306

In [44]:
lr.fit(X_train, y_train)

LinearRegression()

In [45]:
lr.coef_

array([ 3.76790586e+02,  5.09349648e+03,  5.34456279e+03,  4.69593813e+01,
        3.17797423e+01,  4.06292728e+01,  7.32287195e-01, -9.43754948e+02,
        2.18333065e+04])

In [46]:
pd.DataFrame(lr.coef_, index = X.columns,)

0
year_built        376.790586
overall_cond     5093.496476
full_bath        5344.562793
garage_area        46.959381
square_footage     31.779742
lot_frontage       40.629273
lot_area            0.732287
yr_sold          -943.754948
overall_qual    21833.306540

In [47]:
print(f'Training R2: {lr.score(X_train, y_train)}')
print(f'Test R2: {lr.score(X_test, y_test)}')

Training R2: 0.7843234431694472
Test R2: 0.803255298902514


In [48]:
preds = lr.predict(X_test)

print(f'MSE: {mean_squared_error(y_test, preds)}')
print(f'RMSE: {mean_squared_error(y_test, preds, squared = False)}')

MSE: 1332535901.8113594
RMSE: 36503.91625307289


***Findings***

Cross val score - 76.8% giving us a pretty decent model performance score with the features chosen.

R2 Train and Test - 78.4% train and 80.3% test. Showing that the variability for cost in the features makes up the majority of sale price. 

RMSE - average distance from our predicted salesprice was $36,503.92.

## Model 2

In [49]:
X = train[['year_built', 'overall_cond', 'overall_qual', 'full_bath', 'garage_area', 'square_footage', 'bldg_type_2fmCon', 'bldg_type_Duplex', 'bldg_type_Twnhs', 'bldg_type_TwnhsE']]

y = train['saleprice']

In [50]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 44)

In [51]:
lr = LinearRegression()

In [52]:
cross_val_score(lr, X_train, y_train).mean()

0.7705213572408397

After adding overall quality and building type we saw our cross val score go to 77.1%. So the model is becoming better slowly.

In [53]:
lr.fit(X_train, y_train)

LinearRegression()

In [54]:
lr.coef_

array([   403.95211698,   4581.48720463,  21068.86822633,   6510.63328453,
           46.99531852,     33.68764974,  -4499.65691034, -19629.13463294,
       -24181.19119127, -15129.84978021])

In [55]:
pd.DataFrame(lr.coef_, index = X.columns,)

0
year_built          403.952117
overall_cond       4581.487205
overall_qual      21068.868226
full_bath          6510.633285
garage_area          46.995319
square_footage       33.687650
bldg_type_2fmCon  -4499.656910
bldg_type_Duplex -19629.134633
bldg_type_Twnhs  -24181.191191
bldg_type_TwnhsE -15129.849780

In [56]:
print(f'Training R2: {lr.score(X_train, y_train)}')
print(f'Test R2: {lr.score(X_test, y_test)}')

Training R2: 0.786143352043418
Test R2: 0.801549179676049


In [57]:
preds = lr.predict(X_test)

print(f'MSE: {mean_squared_error(y_test, preds)}')
print(f'RMSE: {mean_squared_error(y_test, preds, squared = False)}')

MSE: 1344091308.9422934
RMSE: 36661.85086629279


***Findings***

Cross val score - 77.1% getting slightly better model performance.

R2 Train and Test - went to 78.7% and 80.2%. Showing barely any change from the previous model by adding building type.

RMSE - average distance from our salesprice was increased to $36,661.85. This is pulling us further away from our average which could mean we are overfitting.

## Model 3

In [58]:
X = train[['year_built', 'overall_cond', 'overall_qual', 'full_bath', 'garage_area', 'square_footage', 'bldg_type_2fmCon', 'bldg_type_Duplex', 'bldg_type_Twnhs', 'bldg_type_TwnhsE','bldg_type_1Fam', 'kitchen_qual_Ex', 'kitchen_qual_Fa', 'kitchen_qual_Gd', 'kitchen_qual_TA']]

y = train['saleprice']

In [59]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 44)

In [60]:
lr = LinearRegression()

In [61]:
cross_val_score(lr, X_train, y_train).mean()

0.7974105396449155

By adding kitchen quality to the features we saw another jump in cross value to 79.7%.

In [62]:
lr.fit(X_train, y_train)

LinearRegression()

In [63]:
lr.coef_

array([ 3.99329932e+02,  4.48426683e+03,  1.69298349e+04,  7.96426233e+03,
        4.35009122e+01,  2.94500463e+01,  9.11827397e+03, -4.51272053e+03,
       -1.08026904e+04, -6.43230344e+03,  1.26294404e+04,  4.22556576e+04,
       -1.32846197e+04, -1.01115596e+04, -1.88594783e+04])

In [64]:
pd.DataFrame(lr.coef_, index = X.columns,)

0
year_built          399.329932
overall_cond       4484.266834
overall_qual      16929.834940
full_bath          7964.262331
garage_area          43.500912
square_footage       29.450046
bldg_type_2fmCon   9118.273975
bldg_type_Duplex  -4512.720528
bldg_type_Twnhs  -10802.690438
bldg_type_TwnhsE  -6432.303437
bldg_type_1Fam    12629.440428
kitchen_qual_Ex   42255.657558
kitchen_qual_Fa  -13284.619717
kitchen_qual_Gd  -10111.559556
kitchen_qual_TA  -18859.478285

In [65]:
print(f'Training R2: {lr.score(X_train, y_train)}')
print(f'Test R2: {lr.score(X_test, y_test)}')

Training R2: 0.8122056379951672
Test R2: 0.8409494673761375


In [66]:
preds = lr.predict(X_test)

print(f'MSE: {mean_squared_error(y_test, preds)}')
print(f'RMSE: {mean_squared_error(y_test, preds, squared = False)}')

MSE: 1077236356.258968
RMSE: 32821.2790161957


***Findings***

Cross val score - went up to 79.7% showing that adding kitchens was a decent model feature to add for the overall performance of the model.

R2 Train and Test - 81.2% and 84.1% increase giving an even more solid performance between the last few models.

RMSE - average distance from our salesprice was decreased to $32,821.28. This is a large jump down from the previous model showing a closer distance close than the last model. 

## Chonk

What happens if I just throw all of the numeric features I had available into this.

In [40]:
#numeric_columns = train.select_dtypes(include = ['int64', 'float64'])

#print(numeric_columns)

In [4]:
X = train[['year_built', 'overall_cond', 'overall_qual', 'full_bath', 'garage_area', 'square_footage', 'bldg_type_1Fam', 'bldg_type_2fmCon', 'bldg_type_Duplex', 'bldg_type_Twnhs', 'bldg_type_TwnhsE','kitchen_qual_Ex', 'kitchen_qual_Fa', 'kitchen_qual_Gd', 'kitchen_qual_TA', 'year_remod/add', 'year_built', 'mas_vnr_area', 'lot_area', 'lot_frontage', 'ms_subclass']]

y = train['saleprice']

In [5]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 44)

In [6]:
lr = LinearRegression()

In [7]:
cross_val_score(lr, X_train, y_train).mean()

0.7990773215338296

I just threw every numeric feature I had into this one giving me a 79.9% model score.

In [8]:
lr.fit(X_train, y_train)

LinearRegression()

In [9]:
lr.coef_

array([ 1.74085298e+02,  3.94773087e+03,  1.64763691e+04,  8.14996917e+03,
        3.85086304e+01,  2.56484667e+01,  1.56624342e+04,  6.21176327e+03,
       -2.22306186e+03, -1.41679837e+04, -5.48315188e+03,  4.06808791e+04,
       -1.26702392e+04, -1.00787664e+04, -1.79318735e+04,  1.32908873e+02,
        1.74085298e+02,  3.53364821e+01,  5.12003796e-01, -3.21625389e+01,
        4.26286387e+01])

In [10]:
pd.DataFrame(lr.coef_, index = X.columns,)

0
year_built          174.085298
overall_cond       3947.730874
overall_qual      16476.369145
full_bath          8149.969170
garage_area          38.508630
square_footage       25.648467
bldg_type_1Fam    15662.434204
bldg_type_2fmCon   6211.763269
bldg_type_Duplex  -2223.061858
bldg_type_Twnhs  -14167.983739
bldg_type_TwnhsE  -5483.151876
kitchen_qual_Ex   40680.879067
kitchen_qual_Fa  -12670.239156
kitchen_qual_Gd  -10078.766410
kitchen_qual_TA  -17931.873502
year_remod/add      132.908873
year_built          174.085298
mas_vnr_area         35.336482
lot_area              0.512004
lot_frontage        -32.162539
ms_subclass          42.628639

In [11]:
print(f'Training R2: {lr.score(X_train, y_train)}')
print(f'Test R2: {lr.score(X_test, y_test)}')

Training R2: 0.8190668803805065
Test R2: 0.8524789193187309


In [12]:
preds = lr.predict(X_test)

print(f'MSE: {mean_squared_error(y_test, preds)}')
print(f'RMSE: {mean_squared_error(y_test, preds, squared = False)}')

MSE: 999148313.450121
RMSE: 31609.30738643479


## Lasso and Ridge

Using Chonk to take all the numeric data and throw it into a ridge and lasso to see how tight I could make my already normal R2 scores.

In [13]:
sc = StandardScaler()
Z_train = sc.fit_transform(X_train)
Z_test = sc.transform(X_test)

In [14]:
print(f'Z_train shape is: {Z_train.shape}')
print(f'y_train shape is: {y_train.shape}')
print(f'Z_test shape is: {Z_test.shape}')
print(f'y_test shape is: {y_test.shape}')

Z_train shape is: (1536, 21)
y_train shape is: (1536,)
Z_test shape is: (513, 21)
y_test shape is: (513,)


In [15]:
ols = LinearRegression()

ols.fit(Z_train, y_train)

LinearRegression()

In [16]:
print(f'Training Score: {ols.score(Z_train, y_train)}')
print(f'Testing Score: {ols.score(Z_test, y_test)}')

Training Score: 0.8178970403730074
Testing Score: 0.8546265847304103


In [17]:
ols.coef_

array([-1.14486448e+17,  4.26400000e+03,  2.42205928e+04,  3.95433391e+03,
        8.94877141e+03,  2.18107656e+04, -1.12600625e+18, -4.40153007e+17,
       -5.83422892e+17, -4.83462735e+17, -8.17783074e+17, -1.06784451e+18,
       -6.18818410e+17, -1.98819592e+18, -2.04349453e+18,  3.58488904e+03,
        1.14486448e+17,  6.05468853e+03,  4.36143834e+03, -1.05066954e+03,
        1.83323602e+03])

In [18]:
alpha = 100

alpha * (ols.coef_ ** 2).sum()

1.2382747676260264e+39

In [19]:
from sklearn.linear_model import Ridge

In [20]:
ridge = Ridge(alpha = 999)

ridge.fit(Z_train, y_train)

# Evaluate model using R2.
print(f'Training Score: {ridge.score(Z_train, y_train)}')
print(f'Testing Score: {ridge.score(Z_test, y_test)}')

Training Score: 0.7956305105398384
Testing Score: 0.8074175439367592


By increasing the alpha to 999 I pretty much punished all of the larger coefficients in my model to provide the tightest R2 for Chonk. It dropped a few percentage points and became a more conserative model to showcase future predictions.

***LASSO***

In [21]:
from sklearn.linear_model import Lasso, LassoCV

In [29]:
# Set up a list of Lasso alphas to check.
l_alphas = np.logspace(-3, 0, 100)
# Cross-validate over our list of Lasso alphas.
lasso_cv = LassoCV(alphas = l_alphas, max_iter = 25000)
# Fit model using best ridge alpha!
lasso_cv.fit(Z_train, y_train)

#it went through a lot to get that warning out of the way.

LassoCV(alphas=array([0.001     , 0.00107227, 0.00114976, 0.00123285, 0.00132194,
       0.00141747, 0.00151991, 0.00162975, 0.00174753, 0.00187382,
       0.00200923, 0.00215443, 0.00231013, 0.00247708, 0.00265609,
       0.00284804, 0.00305386, 0.00327455, 0.00351119, 0.00376494,
       0.00403702, 0.00432876, 0.00464159, 0.00497702, 0.0053367 ,
       0.00572237, 0.00613591, 0.00657933, 0.0070548 , 0.00756463,
       0.008...
       0.09326033, 0.1       , 0.10722672, 0.1149757 , 0.12328467,
       0.13219411, 0.14174742, 0.15199111, 0.16297508, 0.17475284,
       0.18738174, 0.2009233 , 0.21544347, 0.23101297, 0.24770764,
       0.26560878, 0.28480359, 0.30538555, 0.32745492, 0.35111917,
       0.37649358, 0.40370173, 0.43287613, 0.46415888, 0.49770236,
       0.53366992, 0.57223677, 0.61359073, 0.65793322, 0.70548023,
       0.75646333, 0.81113083, 0.869749  , 0.93260335, 1.        ]),
        max_iter=25000)

In [23]:
lasso_cv.alpha_

1.0

In [24]:
print(lasso_cv.score(Z_train, y_train))
print(lasso_cv.score(Z_test, y_test))

0.8190668761631288
0.8524760668769925


So after all of that scrolling we can see that LASSO's affect was negligible due to the amount of categorical variables that were changed included in the X. Given that our alpha was a nice 1 nothing was really going to happen as Lasso just zero'd out the majority of my coefficients. 

The model overall isn't suffering from high variance between the testing and training scores so these results for both were as expected. 